### PR TITLE
Prepare repo for writing Proposals

### PR DESCRIPTION
**Pull request by gkatsev**
I thought that the current template was more than good enough to start with, though, I added a note to it to mention Goals and Non-Goals. I think we can iterate on the template as we start writing proposals.

Furthermore, I used rust's RFC process for the submitting a proposal step in the README. One thing to point out there is that, similarly to rust's RFC process, I currently have a full 2-week call of consensus to make sure that folks get a chance to review the proposal before it gets merged as ideally, there aren't major changes done to it once it's merged.

The `proposals` folder has a `.gitkeep` file to keep the folder around in git. The file can be removed once we merge the first proposal.

I think that Discussions/Issues would be useful to getting to whether something is needed and finding a direction, but that most of the work on proposals should happen in a proposals PR in the future. For example, I think we have more than enough traction in the quality levels issue to start on a proposal. Also, a proposal can be merged before it's implemented in either media-chrome or another Web Component. This is so that we could develop a backlog of proposals, especially if they are well-thought-out, without necessarily require that they are implemented before they are merged.

### CODE REVIEW

**Comment by gkatsev** [comment]
With a brief conversation with @cjpillsbury, I was thinking that maybe it'll be useful to have a base video/media element that serves as the canonical implementation of these media-ui-extensions, an `x-video` element. This could also serve as a base-class for custom impementations of these.
Then, the addition to this current proposal is that after a proposal is merged, it'll be in an "unimplemented" stage and once implemented in `x-video`, it could move to an `implemented/` folder or the stage could be incremented. At that point, the proposal could receive an WebIDL definition.
Additionally, `x-video` could serve as infrastructure for a web-platform-tests type thing for media-ui-extensions.

**Comment by gkatsev** [comment]
Could use https://github.com/muxinc/custom-video-element as the `x-video`.

File: README.md
Comment (cjpillsbury) [inline README.md:7]: nitpick(blocking). Grammar.

```suggestion
We want these proposals to be something that we can eventually propose to the Media WG or WhatWG as additional features to the video element and anything related. In the meantime, proposals accepted to media-ui-extensions can be used as a specification to keep things interoperable between implementers.
```

File: README.md
Comment (cjpillsbury) [inline README.md:10]: suggestion(blocking): Ambiguous. Per out of band conversation, it appears that every one of our current [Media UI Extensions Issues](https://github.com/video-dev/media-ui-extensions/issues) should have already been proposals, whereas issues should be reserved for much simpler questions like "Would something that models available renditions be a good candidate for a media ui extension proposal? Having some way to to know details about the available renditions in a predictable way would help for building things like a rendition selector, etc. etc.". Maybe consider adding examples and non-examples?

File: README.md
Comment (cjpillsbury) [inline README.md:16]: suggestion(non-blocking): Maybe call out specifically that this may require a PR followed by an additional commit + push with the rename (🐔 + 🥚 problem)

File: README.md
Comment (gkatsev) [reply README.md:10]: I do think that some ambiguity is warranted here.

Like the Rust RFC, the idea is that someone who isn't part of the core team or deeply involved in video doesn't waste time on a detailed proposal if they aren't sure that it could be useful or that there may be enough community buy in. It's definitely a bit easier on the rust side because it's the core team that decides where-as we're trying to make this more of a community thing.

That said, can still be cleared up a bit with some examples.

I do think that until recently, the renditions issue probably couldn't have been a full proposal (or at least would've been in PR for a long time) because there wasn't a good direction for it. The DVR and stream types probably could've been a PR from the beginning.

File: README.md
Comment (cjpillsbury) [inline README.md:19]: suggestion(non-blocking): do we want to make it a hard minimum of 2 weeks, or should we allow for something more like, 

> The CfC will last a minimum of 2 weeks to make sure that everyone gets a chance to review. If all relevant parties have explicitly finished reviewing before 2 weeks, the CfC may end early.

This may depend on who counts as "everyone" here. If we want to treat this more as an open "request for comments" to the larger community, we'd likely want a clear, fixed time window. If we're instead more concerned about "explicit relevant parties" (which could vary per proposal), my suggested change may be ok.

File: README.md
Comment (cjpillsbury) [inline README.md:23]: suggestion(non-blocking): Should we instead formalize something like a "draft proposal" for an accepted proposal that has yet to be implemented? With our current directory structure and top level readme, it may be difficult to identify superseding proposals and the set of proposals that are
1. accepted
2. implemented
3. not superseded

**Review by cjpillsbury — CHANGES_REQUESTED** [review]
A few asks here. Love getting this process beefed up. (🐮).

**Comment by cjpillsbury** [comment]
idea(non-blocking): Currently the readme is very "video player"/"video element"-centric. As a followup effort, we should probably reword this more generically as "media player"/`HTMLMediaElement` (and then make explicit callouts to audio vs. video where relevant) as an improvement here.

File: README.md
Comment (cjpillsbury) [reply README.md:10]: Ah ok... I think I'm still unclear on what in your mind is a good point for something becoming a proposal PR. Maybe that's fine? Taking the renditions proposal as an example, it was pretty clear (imo) early on that **_some_** kind of API would be beneficial as a media-ui-extension, even if there were a lot of details to work through on what it could/would look like. From our followup conversation, I guess I assumed that once we were confident **_something_** should be proposed, it should move from an issue to a PR. We could also solve this by formalizing the process and (to be defined) relevant parties for requesting an issue be "converted" into a draft proposal PR?

Comment (gkatsev) [reply README.md:10]: I think a proposal is something that you can fill out the template pretty well for, as well as include a specific proposal for how to solve the issue at hand. If you have just a specific problem you're trying to solve, or you haven't thought about alternatives or prior art or stuff like that, it probably belongs as an issue first rather than going directly to a PR.

A lot of the proposals we have coming down the line are things that we have been thinking about for a long time and had a pretty good idea for a direction for them. The fact that they did have a lot of discussion doesn't mean it shouldn't have been a PR proposal.

For the renditions issue, we didn't have a clear direction at first, so, opening an issue was a good call. Now, that we have an idea to proceed, making a PR proposal would be the next step to move the renditions proposal forward.

This section definitely applies less to us since we're likely to only submit a proposal that we need. But this also falls into something else you called out re "the team". Like, who is making the final decision for these things? I was on purpose a bit ambiguous, but maybe too ambiguous. For Rust, it's a lot clearer who it is that makes these decisions. Probably should start with the media-chrome committers (and have them have write access to this repo)? Though, at the moment, I think only heff has write access here.

**Comment by heff** [comment]
This all seems fine to me. @gkatsev want to address some of @cjpillsbury's comments and then we can merge? I think we can clean up language issues and ambiguous things, but probably don't need to stress the exact details too much. I'm sure we'll learn more about what we want as we actually go through the process and we can easily change things as we do.

> probably reword this more generically as "media player"

Sounds good to me.

File: README.md
Comment (gkatsev) [reply README.md:20]: It's not really a new state, it's back to the original state. Just not called out until here.
So, when a PR is made it's in "development mode" and then it gets accepted, rejected, or postponed. The Rust RFC is similarly inexact in only calling it development mode here and not earlier.

I do think there should be ambiguity in terms of completing the CfC, though, happy to reword it as well.

Maybe something like:

> If there is enough consensus, the CfC will close successfully. If there is no consensus, the proposal could be rejected or go back to be re-edited.

Comment (cjpillsbury) [reply README.md:20]: That sounds perfect for this phase.

**Review by cjpillsbury — APPROVED** [review]
😎

**Comment by gkatsev** [comment]
Follow-ups:
- [ ] Add "Implemented" folder or Stages like TC39 proposals and update the process to include these
- [ ] Add https://github.com/muxinc/custom-video-element as the canonical implementation of these proposals, but not the only implementation
- [ ] Update the wording in the README to be more general of "Media Elements"
- [ ] Clarify how to call a CfC (comment on the PR)

**Pull request merged**
